frontend/app/src/components/home/CurrentChatMessages.svelte

```diff
@@ -412,6 +415,19 @@
         client.deleteMessage(chat.chatId, undefined, message.messageId);
     }
 
+    function onUndeleteMessage(ev: CustomEvent<Message>) {
+        doUndeleteMessage(ev.detail);
+    }
+
+    function doUndeleteMessage(message: Message) {
+        if (message.content.kind !== "deleted_content" || 
+            message.content.deletedBy !== user.userId) {
+            return
+        };
+
+        client.undeleteMessage(chat.chatId, undefined, message.messageId);
```

Comment:
We aren't checking whether this worked or not. Generally we would toast a failure (though I can see that deleteMessage does not). Something like `blockUser` in the same file looks slightly better to me.

Reply:
I suppose the reasoning for delete message is that the user can see for themseves that the delete fails because the message reappears. Likewise they can see if the undelete fails because the message returns to the deleted state.

Reply:
I'll add toast for undelete failing and leave delete as is.